src/core/shards/network.cpp

```diff
@@ -335,7 +457,58 @@ struct Server : public NetworkBase {
 };
 
 struct Client : public NetworkBase {
+  std::array<SHExposedTypeInfo, 1> _exposing;
+
+  SHExposedTypesInfo exposedVariables() {
+    _exposing[0].name = "Network.Peer";
+    _exposing[0].help = SHCCSTR("The required peer.");
+    _exposing[0].exposedType = Client::PeerType;
+    return {_exposing.data(), 1, 0};
+  }
+
+  static inline Type PeerType{{SHType::Object, {.object = {.vendorId = CoreCC, .typeId = PeerCC}}}};
+  static inline Type PeerObjectType = Type::VariableOf(PeerType);
+
+  static SHTypesInfo outputTypes() { return PeerType; }
+
   NetworkPeer _peer;
+  ShardsVar _blks{};
+
+  static inline ParamsInfo params = ParamsInfo(
```

Comment:
These need to be in requiredVariables, you can use the `collectRequiredVariables` helper in compose:
```cpp
ExposedInfo _required;
SHTypeInfo compose(const SHInstanceData &data) {
   _required.clear();
   collectRequiredVariables(data.shared, _required, (SHVar&)_addr);
   collectRequiredVariables(data.shared, _required, (SHVar&)_port);
}
```

Reply:
Check the other shards as well since they have similar parameters.

Reply:
But shardsvar is not an actual variable, just shards no?

Reply:
`CoreInfo::ShardsOrNone` nothing to require

Reply:
I meant for _addr and _port

Reply:
Ah good point